app/views/contexts/summary/list_collector_block.py

```diff
@@ -0,0 +1,247 @@
+                return_to="section-summary",
+            )
+
+        driving_question_block = QuestionnaireSchema.get_driving_question_for_list(
+            self._section, summary["for_list"]
+        )
+
+        if driving_question_block:
+            return url_for(
+                "questionnaire.block",
+                block_id=driving_question_block["id"],
+                return_to="section-summary",
+            )
+
+    def _get_related_answers(
+        self, current_list: ListModel
+    ) -> Optional[dict[str, list[Block]]]:
+        section = self._section["id"]
+
+        related_answers = self._schema.get_related_answers_for_list_for_section(
+            section_id=section, list_name=current_list.name
+        )
+        if not related_answers:
+            return None
+
+        related_answers_blocks = {}
+
+        blocks = self.get_blocks_for_related_answers(related_answers)
+
+        for list_id in current_list:
+            serialized_blocks = [
+                Block(
+                    block,
+                    answer_store=self._answer_store,
+                    list_store=self._list_store,
+                    metadata=self._metadata,
+                    response_metadata=self._response_metadata,
+                    schema=self._schema,
+                    location=Location(
+                        list_name=current_list.name,
+                        list_item_id=list_id,
+                        section_id=self._section["id"],
+                    ),
+                    return_to="section-summary",
+                    return_to_block_id=None,
+                ).serialize()
+                for block in blocks
+            ]
+
+            related_answers_blocks[list_id] = serialized_blocks
+
+        return related_answers_blocks
+
+    def _get_item_label(self, list_name: str) -> Optional[str]:
+        for item in self._section["summary"].get("items"):
+            if item["for_list"] == list_name and item.get("item_label"):
+
+                return str(item["item_label"])
+
+    def _get_item_anchor_answer_id(self, list_name: str) -> Optional[str]:
+        for item in self._section["summary"].get("items"):
+            if item["for_list"] == list_name and item.get("item_anchor_answer_id"):
+                return f"#{str(item['item_anchor_answer_id'])}"
+
+    def get_blocks_for_related_answers(
+        self, related_answers: tuple
+    ) -> list[Optional[ImmutableDict]]:
+        blocks = []
+        answers_by_block = defaultdict(list)
+
+        for answer in related_answers:
+            answer_id = answer["identifier"]
+            block = self._schema.get_block_for_answer_id(answer_id)
+
+            block_to_keep = (
+                block["edit_block"]  # type: ignore
+                # block is not optional at this point
+                if block["type"] == "ListCollector"  # type: ignore
+                else block
+            )
```

Comment:
Should type ignore the block itself.

```suggestion
            # block is not optional at this point
            block: Mapping = self._schema.get_block_for_answer_id(answer_id) # type: ignore

            block_to_keep = (
                block["edit_block"]
                if block["type"] == "ListCollector"
                else block
            )
```